lib/src/source_visitor.dart

```diff
@@ -2559,6 +2613,30 @@ class SourceVisitor extends ThrowingAstVisitor {
     builder.unindent();
   }
 
+  @override
+  void visitSwitchPatternCase(SwitchPatternCase node) {
+    _visitLabels(node.labels);
+
+    token(node.keyword);
+    space();
+
+    builder.startBlockArgumentNesting();
+    builder.nestExpression();
+    visit(node.guardedPattern.pattern);
+    builder.unnest();
+    builder.endBlockArgumentNesting();
+
+    visit(node.guardedPattern.whenClause);
+    token(node.colon);
+
+    builder.indent();
+    // TODO(rnystrom): Allow inline cases?
```

Comment:
Heh, this was copy/pasted from `visitBinaryExpression()`. It's probably close to a decade old. :)

I'll just remove the TODOs. I am interested in allowing cases with inline bodies since pattern matching makes that more useful, but I probably don't need a TODO to remind me.